File: backend/app/integrations/github/review_publisher.py
from app.integrations.github.client import GitHubClient


class ReviewPublisher:
    """GitHub review/comment publisher with shared HTTP client."""

    def __init__(self, github_token: str):
        self.github_token = github_token
        self._client = GitHubClient(github_token)

    def create_review(self, owner: str, repo: str, pr_number: int, *, commit_id: str, body: str, comments: list[dict]):
        url = f'/repos/{owner}/{repo}/pulls/{pr_number}/reviews'
        payload = {'commit_id': commit_id, 'body': body, 'event': 'COMMENT', 'comments': comments}
        r = self._client.http.post(url, json=payload)
        return r.json()

    def create_review_comment(self, owner: str, repo: str, pr_number: int, *, commit_id: str, path: str, line: int, body: str, side: str = 'RIGHT'):
        url = f'/repos/{owner}/{repo}/pulls/{pr_number}/comments'
        payload = {'body': body, 'commit_id': commit_id, 'path': path, 'line': int(line), 'side': side}
        r = self._client.http.post(url, json=payload)
        return r.json()

    def post_issue_comment(self, owner: str, repo: str, pr_number: int, body: str):
        url = f'/repos/{owner}/{repo}/issues/{pr_number}/comments'
        r = self._client.http.post(url, json={'body': body})
        return r.json()

File: backend/src/integrations/github/review-publisher.ts
import { GitHubClient } from './client.js';

/**
 * GitHub review/comment publisher with shared HTTP client.
 */
export class ReviewPublisher {
  private client: GitHubClient;

  constructor(githubToken: string) {
    this.client = GitHubClient.getInstance(githubToken);
  }

  async createReview(
    owner: string, repo: string, prNumber: number,
    options: { commitId: string; body: string; comments: Record<string, unknown>[] }
  ): Promise<Record<string, unknown>> {
    const url = `/repos/${owner}/${repo}/pulls/${prNumber}/reviews`;
    const payload = {
      commit_id: options.commitId,
      body: options.body,
      event: 'COMMENT',
      comments: options.comments,
    };
    const r = await this.client.http.post(url, { json: payload as Record<string, unknown> });
    return r.data;
  }

  async createReviewComment(
    owner: string, repo: string, prNumber: number,
    options: { commitId: string; path: string; line: number; body: string; side?: string }
  ): Promise<Record<string, unknown>> {
    const url = `/repos/${owner}/${repo}/pulls/${prNumber}/comments`;
    const payload = {
      body: options.body,
      commit_id: options.commitId,
      path: options.path,
      line: options.line,
      side: options.side || 'RIGHT',
    };
    const r = await this.client.http.post(url, { json: payload as Record<string, unknown> });
    return r.data;
  }

  async postIssueComment(
    owner: string, repo: string, prNumber: number, body: string
  ): Promise<Record<string, unknown>> {
    const url = `/repos/${owner}/${repo}/issues/${prNumber}/comments`;
    const r = await this.client.http.post(url, { json: { body } });
    return r.data;
  }
}
